src/libraries/System.Private.CoreLib/src/System/Random.cs

```diff
@@ -1,197 +1,599 @@
-        }
+            public abstract int Next();
 
-        public virtual double NextDouble() => Sample();
+            public abstract int Next(int maxValue);
 
-        public virtual void NextBytes(byte[] buffer)
+            public abstract int Next(int minValue, int maxValue);
+
+            public abstract long NextInt64();
+
+            public abstract long NextInt64(long maxValue);
+
+            public abstract long NextInt64(long minValue, long maxValue);
+
+            public abstract double NextDouble();
+
+            public abstract float NextSingle();
+
+            public abstract void NextBytes(byte[] buffer);
+
+            public abstract void NextBytes(Span<byte> buffer);
+        }
+
+        #region Implementations
+        /// <summary>
+        /// Provides an implementation of the xoshiro256** algorithm. This implementation is used
+        /// when no seed is specified and an instance of the base Randomclass is constructed.  As
+        /// such, we are free to implement however we see fit, withoutback compat concerns around
+        /// the sequence of numbers generated or what methods call what other methods.
+        /// </summary>
+        private sealed class Xoshiro256StarStarImpl : ImplBase
         {
-            if (buffer is null)
+            // NextUInt64 is based on the algorithm from http://prng.di.unimi.it/xoshiro256starstar.c:
+            //
+            //     Written in 2018 by David Blackman and Sebastiano Vigna (vigna@acm.org)
+            //
+            //     To the extent possible under law, the author has dedicated all copyright
+            //     and related and neighboring rights to this software to the public domain
+            //     worldwide. This software is distributed without any warranty.
+            //
+            //     See <http://creativecommons.org/publicdomain/zero/1.0/>.
+
+            private ulong _s0, _s1, _s2, _s3;
+
+            public unsafe Xoshiro256StarStarImpl()
             {
-                ThrowHelper.ThrowArgumentNullException(ExceptionArgument.buffer);
+                do
+                {
+                    Guid g;
+
+                    // Seed the four ulongs (4 64-bit values) from two Guids (2 128-bit values)
+                    g = Guid.NewGuid();
+                    _s0 = *(ulong*)&g;
+                    _s1 = *((ulong*)&g + 1);
+
+                    g = Guid.NewGuid();
+                    _s2 = *(ulong*)&g;
+                    _s3 = *((ulong*)&g + 1);
+                }
+                while ((_s0 | _s1 | _s2 | _s3) == 0); // at least one value must be non-zero
             }
 
-            for (int i = 0; i < buffer.Length; i++)
+            /// <summary>Produces a value in the range [0, ulong.MaxValue].</summary>
+            private ulong NextUInt64()
             {
-                buffer[i] = (byte)InternalSample();
+                ulong result = BitOperations.RotateLeft(_s1 * 5, 7) * 9;
```

Comment:
Can confirm this method matches `xoshorio256**` so I think the "interesting" part of this PR is how smaller values are extracted..